appserver/security/core-ee/src/main/java/com/sun/enterprise/security/auth/realm/jdbc/JDBCRealm.java

```diff
@@ -139,37 +141,31 @@ public synchronized void init(Properties props) throws BadRealmException, NoSuch
         String groupTable = props.getProperty(PARAM_GROUP_TABLE);
         String groupNameColumn = props.getProperty(PARAM_GROUP_NAME_COLUMN);
         String groupTableUserNameColumn = props.getProperty(PARAM_GROUP_TABLE_USER_NAME_COLUMN, userNameColumn);
-        cr = (ActiveDescriptor<ConnectorRuntime>) Util.getDefaultHabitat()
-                .getBestDescriptor(BuilderHelper.createContractFilter(ConnectorRuntime.class.getName()));
+
+        connectorRuntimeDescriptor = (ActiveDescriptor<ConnectorRuntime>)
+            getDefaultHabitat().getBestDescriptor(BuilderHelper.createContractFilter(ConnectorRuntime.class.getName()));
 
         if (jaasCtx == null) {
-            String msg = sm.getString("realm.missingprop", BaseRealm.JAAS_CONTEXT_PARAM, "JDBCRealm");
-            throw new BadRealmException(msg);
+            throw new BadRealmException(String.format(MISSING_PROPERTY, JAAS_CONTEXT_PARAM, "JDBCRealm"));
         }
 
         if (dsJndi == null) {
-            String msg = sm.getString("realm.missingprop", PARAM_DATASOURCE_JNDI, "JDBCRealm");
-            throw new BadRealmException(msg);
+            throw new BadRealmException(String.format(MISSING_PROPERTY, PARAM_DATASOURCE_JNDI, "JDBCRealm"));
```

Comment:
Here an below in six places `MISSING_PROPERTY` constant uses `{...}` placeholders

Reply:
Thanks again, I think I now finally got them all!